docker-entrypoint.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -e
+
+# setup ros environment
+source "/opt/ros/noetic/setup.bash"
+if [ -f "/code/software/devel/setup.bash" ]; then
```

Comment:
If you'd like to use `colcon build` in the compile command, change `devel` to `install` here

Reply:
I tried to switch this to `colcon` but ran into issues - I'll save that for a different PR